packages/core/src/modules/sd-jwt-vc/credential-status/token-status-list/TokenStatusList.ts

```diff
@@ -0,0 +1,237 @@
+import { StatusList, getListFromStatusListJWT } from '@sd-jwt/jwt-status-list'
+import { isURL } from 'class-validator'
+import { injectable } from 'tsyringe'
+import { AgentContext } from '../../../../agent'
+import { JwsService, Jwt, JwtPayload } from '../../../../crypto'
+import { CredoError } from '../../../../error'
+import { isDid } from '../../../../utils'
+import { DidsApi, getPublicJwkFromVerificationMethod, parseDid } from '../../../dids'
+import { SdJwtVcModuleConfig } from '../../SdJwtVcModuleConfig'
+import { SdJwtVcIssuer } from '../../SdJwtVcOptions'
+import { SdJwtVcService } from '../../SdJwtVcService'
+import { TokenStatusListError } from './TokenStatusListError'
+import {
+  PublishTokenStatusListOptions,
+  TokenStatusListJwtPayload,
+  TokenStatusListRegistry,
+} from './TokenStatusListRegistry'
+
+export interface CreateTokenStatusListOptions extends PublishTokenStatusListOptions {
+  size: number
+  publish: boolean
+  bitsPerStatus?: number
+}
+
+export interface CreateTokenStatusListResult {
+  jwt: string
+  uri?: string
+}
+
+export interface RevokeIndicesOptions extends PublishTokenStatusListOptions {
+  uri: string
+  indices: number[]
+  publish: boolean
+}
+
+/**
+ * @internal
+ */
+@injectable()
+export class TokenStatusListService {
```

Comment:
Thinking whether we should make this an external module, and not depend it on SD-jWT VC, but for now it's fine i think.